From 7973bb1a52c6bdccd535ea7a59714cdba2eb438b Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Wed, 27 Mar 2013 08:46:13 +0900
Subject: [PATCH] Policy: install-info is run by a dpkg trigger.

Wording: Jonathan Nieder <jrnieder@gmail.com>
Seconded: Charles Plessy <plessy@debian.org>
Seconded: Russ Allbery <rra@debian.org>
Closes: #669915
---
 policy.sgml | 38 +++++++++++++++++++++-----------------
 1 file changed, 21 insertions(+), 17 deletions(-)

diff --git a/policy.sgml b/policy.sgml
index a41bc1f..2bafbd6 100644
--- a/policy.sgml
+++ b/policy.sgml
@@ -1330,9 +1330,9 @@ zope.
 	  The package installation scripts should avoid producing
 	  output which is unnecessary for the user to see and
 	  should rely on <prgn>dpkg</prgn> to stave off boredom on
-	  the part of a user installing many packages.  This means,
-	  amongst other things, using the <tt>--quiet</tt> option on
-	  <prgn>install-info</prgn>.
+	  the part of a user installing many packages. This means,
+	  amongst other things, not passing the <tt>--verbose</tt>
+	  option to <prgn>update-alternatives</prgn>.
 	</p>
 
 	<p>
@@ -3969,8 +3969,7 @@ Checksums-Sha256:
 	  Programs called from maintainer scripts should not normally
 	  have a path prepended to them. Before installation is
 	  started, the package management system checks to see if the
-	  programs <prgn>ldconfig</prgn>,
-	  <prgn>start-stop-daemon</prgn>, <prgn>install-info</prgn>,
+	  programs <prgn>ldconfig</prgn>, <prgn>start-stop-daemon</prgn>,
 	  and <prgn>update-rc.d</prgn> can be found via the
 	  <tt>PATH</tt> environment variable. Those programs, and any
 	  other program that one would expect to be in the
@@ -10497,18 +10496,23 @@ name ["<var>syshostname</var>"]:
 
 	<p>
 	  The <prgn>install-info</prgn> program maintains a directory of
-	  installed info documents in <file>/usr/share/info/dir</file> for
-	  the use of info readers.<footnote>
-	    It was previously necessary for packages installing info
-	    documents to run <prgn>install-info</prgn> from maintainer
-	    scripts.  This is no longer necessary.  The installation
-	    system now uses dpkg triggers.
-	  </footnote>
-	  This file must not be included in packages.  Packages containing
-	  info documents should depend on <tt>dpkg (>= 1.15.4) |
-	  install-info</tt> to ensure that the directory file is properly
-	  rebuilt during partial upgrades from Debian 5.0 (lenny) and
-	  earlier.
+	  installed info documents in <file>/usr/share/info/dir</file> for the
+	  use of info readers.  This file must not be included in packages
+	  other than <package>install-info</package>.
+	</p>
+
+	<p>
+	  <prgn>install-info</prgn> is automatically invoked when
+	  appropriate using dpkg triggers.  Packages other than
+	  <package>install-info</package> <em>should not</em> invoke
+	  <prgn>install-info</prgn> directly and <em>should not</em>
+	  depend on, recommend, or suggest <package>install-info</package>
+	  for this purpose.
+	</p>
+
+	<p>
+	  Info readers requiring the <file>/usr/share/info/dir</file> file
+	  should depend on <package>install-info</package>.
 	</p>
 
 	<p>
-- 
1.8.2.rc0

